bundle/edu.gemini.spModel.core/src/main/scala/edu/gemini/spModel/core/package.scala

```diff
@@ -8,4 +11,30 @@ package object core {
   type Dec = Declination
   val  Dec = Declination
 
-}
\ No newline at end of file
+  type Ephemeris = Long ==>> Coordinates
+
+  /** Operations for maps of types that we can interpolate.  We use these for Ephemerides. */ 
+  implicit class MoreMapOps[K, V](m: K ==>> V)(implicit O: Order[K], I: Interpolate[K,V]) {
+
+    /** Perform an exact or interpolated lookup. */
+    def iLookup(k: K): Option[V] =
+      m.lookup(k) match {
+        case Some(v) => Some(v)
+        case None    => 
+          val (lt, gt) = m.split(k)
+          ^(lt.findMax, gt.findMin)(I.interpolate(_, _, k))
```

Comment:
It's syntax for `lift2` ... `^(a, b)(f)` is equivalent to `(a |@| b)(f)`. I'll switch it to the latter since we use that one more.